Automattic/wp-calypso
# Staging Sites: Display error when production->staging synchronization fails

Related to https://github.com/Automattic/dotcom-forge/issues/4271

## Proposed Changes

In this PR, I propose to display an error when production<->staging synchronization fails:

![Screenshot 2023-10-27 at 15 50 04](https://github.com/Automattic/wp-calypso/assets/727413/b0a55ffc-7704-4c94-b7a0-25fa1e646ab1)

## Testing Instructions



1. Create a site with a Business plan
2. Navigate to Settings -> Hosting Configuration
3. Add a staging site
4. Using `wpsh`, fill meta to simulate failed synchronization:
```
update_site_meta( PRODUCTION_BLOG_ID, 'site_staging_sync_data', array( 'status' => 'failed') )
```
5. Refresh Hosting Configuration
6. Confirm that error is displayed with CTA asking customer to contact support

## Pre-merge Checklist
